accumulator/utils.go

```diff
@@ -10,6 +10,15 @@ var verbose = false
 
 // takes a slice of dels, removes the twins (in place) and returns a slice
 // of parents of twins
+//
+// Example with 15-leaf tree in printout.txt line 21:
+// If deleting the nodes 4 through 9, the function would be called with
+// nodes=[4,5,6,7,8] and row=4.
```

Comment:
row would be 3 since rows are counted from 0. So the 15-leaf tree in printout.txt would be 3. This can be seen with pollard.rows().

Reply:
I think row is forestRows here. forestRows of the tree I mentioned is 4. Do you agree?

Reply:
This is what I tried:

```
diff --git a/accumulator/forest_test.go b/accumulator/forest_test.go
index a1595c3..f650ef1 100644
--- a/accumulator/forest_test.go
+++ b/accumulator/forest_test.go
@@ -53,7 +53,7 @@ func TestForestAddDel(t *testing.T) {
 
 func TestForestFixed(t *testing.T) {
        f := NewForest(nil, false)
-       numadds := 5
+       numadds := 15
        numdels := 3
        adds := make([]Leaf, numadds)
        dels := make([]uint64, numdels)
@@ -71,6 +71,7 @@ func TestForestFixed(t *testing.T) {
        if err != nil {
                t.Fatal(err)
        }
+       panic(f.rows)
        fmt.Print(f.ToString())
        fmt.Print(f.PrintPositionMap())
        _, err = f.Modify(nil, dels)
```

It paniced with 4.

I am a bit confused. What are you proposing?

Reply:
Ah oops yeah you're right the height is 4. I forgot that the allocated rows is the one that gets returned.

But for future reference I should mention that the height is 4 not because you count from 1 but because the allocated tree always has 2^n leaves. The tree at line 21 in printout.txt is really this:

```
em
|-------------------------------\
28                              em
|---------------\               |---------------\
24              25              26              em
|-------\       |-------\       |-------\       |-------\
16      17      18      19      20      21      22      em
|---\   |---\   |---\   |---\   |---\   |---\   |---\   |---\
00  01  02  03  04  05  06  07  08  09  10  11  12  13  14  em
```

The tree just didn't list the empty nodes.